client/state/sites/test/selectors.js

```diff
@@ -1253,4 +1274,793 @@ describe( 'selectors', () => {
+			const canManage = canJetpackSiteManage( stateWithNoItems, nonExistingSiteId );
+			expect( canManage ).to.equal( null );
+		} );
+
+		it( 'it should return `null` for a non jetpack site', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: false
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( null );
+		} );
+
+		it( 'it should return `true` if jetpack version is strictly less than 3.4', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					URL: 'https://jetpacksite.me',
+					jetpack: true,
+					options: {
+						jetpack_version: '3.3'
+					}
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( true );
+		} );
+
+		it( 'it should return `true` if the modules has not yet been fetched', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					URL: 'https://jetpacksite.me',
+					jetpack: true,
+					options: {
+						active_modules: null,
+						jetpack_version: '3.4'
+					}
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( true );
+		} );
+
+		it( 'it should return `true` if jetpack version is greater or equal to 3.4 and the manage module is active', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					URL: 'https://jetpacksite.me',
+					jetpack: true,
+					options: {
+						active_modules: [ 'manage' ],
+						jetpack_version: '3.4'
+					}
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( true );
+		} );
+
+		it( 'it should return `false` if jetpack version is greater or equal to 3.4 and the manage module is not active', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					URL: 'https://jetpacksite.me',
+					jetpack: true,
+					options: {
+						active_modules: [ 'sso' ],
+						jetpack_version: '3.4'
+					}
+				}
+			} );
+
+			const canManage = canJetpackSiteManage( state, siteId );
+			expect( canManage ).to.equal( false );
+		} );
+	} );
+
+	describe( '#canJetpackSiteUpdateFiles()', () => {
+		it( 'should return `null` for a non-existing site', () => {
+			const canUpdateFiles = canJetpackSiteUpdateFiles( stateWithNoItems, nonExistingSiteId );
+			expect( canUpdateFiles ).to.equal( null );
+		} );
+
+		it( 'it should return `false` for a non jetpack site', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: false,
+				}
+			} );
+
+			const canUpdateFiles = canJetpackSiteUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( null );
+		} );
+
+		it( 'it should return `false` if jetpack version is smaller than minimum version', () => {
+			const smallerVersion = '3.2';
```

Comment:
We should probably stub the config value ` "jetpack_min_version": "3.3",` so this test can make sense in context and won't break when we update the config value